qiskit/pulse/schedule.py

```diff
@@ -1021,3 +1001,26 @@ def _check_nonnegative_timeslot(timeslots):
                 raise PulseError(
                     "An instruction on {} has a negative "
                     " starting time.".format(chan))
+
+def _reduce_to_stack(schedule):
+    """Loops through pulse instructions and removes delays and zero phase shifts.
+    Args:
+        schedule: Schedule to insert.
+    Returns:
+        A tuple of pulse instructions with delays and zero phase shifts removed.
+    """
+
+    new_schedule = []
+    for inst in schedule.instructions:
+        time, action = inst
```

Comment:
This is more direct:
```suggestion
    for time, inst in schedule.instructions:
```